### PR TITLE
Added quick roll dice buttons, and added scroll to bottom when sending a message

### PR DESCRIPTION
**Pull request by rrgeorge**

### CODE REVIEW

File: src/app/core/message-list/message-list.component.ts [hunk 1]
```diff
@@ -34,6 +35,79 @@ export class MessageListComponent implements OnInit {
...
+        let rollStr = ""
+        let rollRE = /^(\/r(oll)? )?(([0-9]+)[dD]([0-9]+)|0)?((kh|kl)1?)?((\+|\-)[0-9]+)? ?(.*)?/;
+        let m = rollRE.exec(this.messageInput);
+        if (this.messageInput != "" && m == null) {
+            return;
+        }
+        rollStr = (m&&m[1])? m[1] : "/roll ";
+
+        let mod=0
+        if (m&&m[9]) {
```
Comment (Rycochet) [inline src/app/core/message-list/message-list.component.ts]: There's another way of doing this with optional chaining -
```suggestion
        if (m.?[9]) {
```

File: src/app/core/message-list/message-list.component.ts [hunk 2]
```diff
@@ -34,6 +35,79 @@ export class MessageListComponent implements OnInit {
     this.sendMessage();
   }
 
+    quickRoll(r) {
+        let rollStr = ""
+        let rollRE = /^(\/r(oll)? )?(([0-9]+)[dD]([0-9]+)|0)?((kh|kl)1?)?((\+|\-)[0-9]+)? ?(.*)?/;
+        let m = rollRE.exec(this.messageInput);
```
Comment (Rycochet) [inline src/app/core/message-list/message-list.component.ts]: Tbh I'd suggest deconstructing this so it's easier to play with, and also using non-capturing groups so it doesn't care about things that are filler -

```suggestion
        const rollRE = /^(?:\/r(?:oll)? )?(?:([0-9]+)[dD]([0-9]+)|0)?(?:(kh|kl)1?)?([\+\-]?[0-9]+)? ?(.*)?/;
        const m = rollRE.exec(this.messageInput);
        
        // return if no m
        
        const [ num, dice, keep, plus, extra ] = m;
```

Then don't touch `m` itself directly again :-)

File: src/app/core/message-list/message-list.component.ts (hunk 1 above)
Comment (rrgeorge) [reply src/app/core/message-list/message-list.component.ts]: Thanks, didn’t know about that trick.

File: src/app/core/message-list/message-list.component.ts (hunk 2 above)
Comment (rrgeorge) [reply src/app/core/message-list/message-list.component.ts]: That would certainly clean things up. I’ll look into it. Thanks.

**Pull request merged**